### ISSUE TITLE
The proportion of POI IDs in the test set hints

### ISSUE THREAD

**Issue by MAOshangh** [neolifer/LLM4POI#16]
Great job! I tried to reproduce your work, and when analyzing the proportion of answer POI IDs in the test set prompts, I found that the results differed significantly from those in the experiment, with only 35.80%. Here, when extracting features from trajectories, I truncated some overly long trajectories due to memory limitations, not sure if this is the reason, but it is very different from my expectations. So I would like to ask if you did any other preprocessing when handling trajectories, and I am very much looking forward to your reply.

**Comment by neolifer** [comment]
Hi there,

I don't have the original checkpoints anymore. But I've tested with our v2 implementation, which we just released. For the model trained on NYC and tested on NYC, almost all correct datapoints contain the answer in the input prompt, with 50 historical trajectories.

**Comment by MAOshangh** [comment]
Thank you for your reply! So, when you perform similar trajectory retrieval, is the encoder of the pre-trained model fine-tuned by you again, or do you use the original pre-trained LLM encoder?

**Comment by neolifer** [comment]
> Thank you for your reply! So, when you perform similar trajectory retrieval, is the encoder of the pre-trained model fine-tuned by you again, or do you use the original pre-trained LLM encoder?

It was intended to be just the pre-trained LLM encoder, though I realized afterward that using embedding model APIs might have been better. There was a bug in the original code that caused the similarity-retrieved trajectories to be ignored, so the final model essentially used only the historical trajectories. The related ablation study might also be incorrect.

**Comment by MAOshangh** [comment]
Okay, I understand. Thank you very much for your patient reply.
